src/pages/WorkspaceDetails/EditorTab/index.tsx

```diff
@@ -214,12 +213,19 @@ export class EditorTab extends React.PureComponent<Props, State> {
     this.setState({ isDevfileValid: true });
   }
 
-  private onDevfileChange(devfile: che.WorkspaceDevfile, isValid: boolean): void {
+  private onDevfileChange(newValue: string, isValid: boolean): void {
     this.setState({ isDevfileValid: isValid });
     if (!isValid) {
       this.setState({ hasChanges: false });
       return;
     }
+    let devfile: che.WorkspaceDevfile;
+    try {
+      devfile = safeLoad(newValue);
+    } catch (e) {
+      console.error('Devfile parse error', e);
+      return;
+    }
     if (this.areEqual(this.props.workspace.devfile as che.WorkspaceDevfile, devfile)) {
```

Comment:
I wonder if now we should start comparing string value somehow, to allow users cancel their empty space changes, like my:
![Screenshot_20210527_163938](https://user-images.githubusercontent.com/5887312/119836376-5f09fb80-bf0a-11eb-95e4-07eb77d9f9b2.png)
I've created a separate issue for that https://github.com/eclipse/che/issues/19882